platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -64,47 +68,157 @@
       NumberUtils.toInt(System.getProperty("solr.cloud.maxShardPerNode"), 2);
 
+  private static final String ALIAS_PROP = "alias";
+
   @Override
   public org.codice.solr.client.solrj.SolrClient newClient(String core) {
+    return newClient(core, null);
+  }
+
+  @Override
+  public org.codice.solr.client.solrj.SolrClient newClient(
+      String collection, Map<String, Object> properties) {
+
     String zookeeperHosts = System.getProperty("solr.cloud.zookeeper");
+    checkConfig(zookeeperHosts);
 
-    if (StringUtils.isBlank(zookeeperHosts)) {
-      LOGGER.warn(
-          "Cannot create SolrCloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
-      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
-    }
     LOGGER.debug(
-        "Solr({}): Creating a SolrCloud client using Zookeeper hosts [{}]", core, zookeeperHosts);
+        "Solr({}): Creating a Solr Cloud client with configuration using Zookeeper hosts [{}]",
+        collection,
+        zookeeperHosts);
+
+    String alias = properties == null ? null : (String) properties.getOrDefault(ALIAS_PROP, null);
+
     return new SolrClientAdapter(
-        core,
+        collection,
         () ->
             AccessController.doPrivileged(
                 (PrivilegedAction<SolrClient>)
                     () -> {
-                      return createSolrCloudClient(zookeeperHosts, core);
+                      return createSolrCloudClient(zookeeperHosts, collection, alias);
                     }));
   }
 
   @VisibleForTesting
-  SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
+  void addCollectionToAlias(String alias, String collection, CloudSolrClient client) {
+    if (StringUtils.isBlank(alias) || StringUtils.isBlank(collection)) {
+      return;
+    }
+    try {
+      List<String> aliasedCollections = getCollectionsInAlias(client, alias);
+      if (!aliasedCollections.contains(collection)) {
+        List<String> collections = new ArrayList<>();
+        collections.addAll(aliasedCollections);
+        collections.add(collection);
+        collections.addAll(getCollectionsNotInAlias(client, aliasedCollections, alias));
+        createAlias(client, collections, alias);
+      }
+    } catch (SolrServerException | IOException e) {
+      LOGGER.warn("Failed to update alias [{}]", alias, e);
+    }
+  }
+
+  private List<String> getCollectionsInAlias(CloudSolrClient client, String alias)
+      throws SolrServerException, IOException {
+    CollectionAdminResponse aliasResponse =
+        new CollectionAdminRequest.ListAliases().process(client);
+
+    if (aliasResponse != null) {
+      Map<String, String> aliases = aliasResponse.getAliases();
+      if (aliases != null && aliases.containsKey(alias)) {
+        String[] collections = aliases.get(alias).split(",");
+        return Arrays.asList(collections);
+      }
+    }
```

Comment:
✏️ This would be a great place to use Optional if you prefer to have the null checks done for you, eg:
```java
return Optional.ofNullable(aliasResponse)
    .map(CollectionAdminResponse::getAliases)
    .map(a -> a.get(alias))
    .map(ca -> ca.split(",")
    .map(Arrays::asList)
    .orElse(Collections.emptyList());
```